### PR TITLE
Support the unpack operator in signature

### LINKED ISSUE #41286: taskflow does not understand * (argument unpack operator): ValueError: The key 'how' in args is a part of kwargs and therefore reserved.
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.2

### What happened?

Executing the attached repro in airflow results in a runtime error 

ValueError: The key 'how' in args is a part of kwargs and therefore reserved.

### What you think should happen instead?

This dag should execute properly

### How to reproduce

```
from airflow.decorators import dag, task
@dag(
    schedule=None,
    catchup=False,
    tags=["example"],
)
def demo_bug():
    @task()
    def input1() -> str:
        return "A"
    
    @task()
    def input2() -> str:
        return "B"
    
    @task()
    def bug(*str, how="cow"):
        pass

    bug(input1, input2, how="now")

demo_bug()
```

### Operating System

Ubuntu 22.04.4 LTS

### Versions of Apache Airflow Providers

None relevant

### Deployment

Virtualenv installation

### Deployment details

Nothing of note

### Anything else?

All the time

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jabbera**
Support the unpack operator in operator signature.

Closes: #41286

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/operators/test_python.py [hunk 1]
```diff
@@ -272,6 +272,18 @@ def func(dag):
         with pytest.raises(ValueError, match=error_message):
             ti.run()
 
+    def test_unpack_operator_kwargs(self):
+        """Makes sure that keyword arguments are not continually processed after an unpack operator."""
+
+        error_message = "Should be run"
+
+        def func(*unpacked_args, keyword_only_arg):
+            raise RuntimeError(error_message)
```
Comment (jabbera) [inline tests/operators/test_python.py]: This could be a pass, but it doesn't actually ensure the method is actually run.

**Comment by romsharon98** [comment]
The changes look good — great work for a first PR! 😉
I think better add tests [here](https://github.com/apache/airflow/blob/463e28b9ce6b58a64795c96ae7b94aba37770c93/tests/utils/test_operator_helpers.py), let me know what you think.

**Comment by jabbera** [comment]
> I think better add tests [here](https://github.com/apache/airflow/blob/463e28b9ce6b58a64795c96ae7b94aba37770c93/tests/utils/test_operator_helpers.py), let me know what you think.

@romsharon98 test_operator_helpers doesn't have the create_ti function/infrastructure that's available in test_python. I could put the describe method under test directly in test_operator_helpers instead of actually executing a task if that's what you mean?

**Comment by romsharon98** [comment]
> > I think better add tests [here](https://github.com/apache/airflow/blob/463e28b9ce6b58a64795c96ae7b94aba37770c93/tests/utils/test_operator_helpers.py), let me know what you think.
> 
> @romsharon98 test_operator_helpers doesn't have the create_ti function/infrastructure that's available in test_python. I could put the determine method under test directly in test_operator_helpers instead of actually executing a task if that's what you mean?

Yes, this is what I ment.

**Comment by jabbera** [comment]
@romsharon98 Sorry for the delay. This is all et.

**Comment by romsharon98** [comment]
changes looks good!
static tests failed, better working with [pre-commit](https://github.com/apache/airflow/blob/main/contributing-docs/03_contributors_quick_start.rst#configuring-pre-commit) and prevent those errors.

File: airflow/utils/operator_helpers.py [hunk 2]
```diff
@@ -163,6 +163,13 @@ def determine(
         has_wildcard_kwargs = any(p.kind == p.VAR_KEYWORD for p in signature.parameters.values())
 
         for name in itertools.islice(signature.parameters.keys(), len(args)):
+            if signature.parameters[name].kind in [
+                inspect.Parameter.KEYWORD_ONLY,
+                inspect.Parameter.VAR_KEYWORD,
+            ]:
+                # this is only checking positional arguments
+                continue
```
Comment (uranusjr) [inline airflow/utils/operator_helpers.py]:
```suggestion
        for name, param in itertools.islice(signature.parameters.items(), len(args)):
            # Keyword-only arguments can't be passed positionally and are not checked.
            if param.kind == inspect.Parameter.KEYWORD_ONLY:
                continue
            if param.kind == inspect.Parameter.VAR_KEYWORD:
                continue
```

**Review by uranusjr — APPROVED** [review]

File: tests/utils/test_operator_helpers.py [hunk 3]
```diff
@@ -224,3 +224,8 @@ def func(ds_nodash):
         kwargs_callable(*args, **kwargs)
 
     assert "ds_nodash" in str(exc_info)
+
+
+def test_args_and_kwargs_conflicts():
+    kwargs_result = operator_helpers.determine_kwargs(callable10, args=[1, 2], kwargs={"ds_nodash": 1})
```
Comment (uranusjr) [inline tests/utils/test_operator_helpers.py]: We should add a test for `**kwargs`. Maybe use `parametrize` to add `callable11`?

**Review by romsharon98 — APPROVED** [review]

**Comment by jabbera** [comment]
Hi, is there anything else needed for this PR? Thanks!

**Pull request merged**